daemon/volumes.go

```diff
@@ -205,11 +237,26 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 		}
 
 		if len(bind.Name) > 0 && len(bind.Driver) > 0 {
+			// set the label
+			if err := label.SetFileCreateLabel(container.MountLabel); err != nil {
+				return fmt.Errorf("Unable to setup default labeling for volume creation %s: %v", bind.Source, err)
+			}
+
+			// create the volume
 			v, err := createVolume(bind.Name, bind.Driver)
 			if err != nil {
+				// reset the label
+				if e := label.SetFileCreateLabel(""); e != nil {
+					logrus.Errorf("Unable to reset labeling for volume creation %s: %v", bind.Source, e)
```

Comment:
Instead of logging should this just be handled by the caller? We can wrap the returned error with this added context.